samcli/lib/providers/api_provider.py

```diff
@@ -51,44 +48,43 @@ def get_all(self):
 
         yield self.api
 
-    def _extract_api(self, resources):
+    def _extract_api(self):
         """
         Extracts all the routes by running through the one providers. The provider that has the first type matched
         will be run across all the resources
 
         Parameters
         ----------
-        resources: dict
-            The dictionary containing the different resources within the template
         Returns
         ---------
         An Api from the parsed template
         """
 
         collector = ApiCollector()
-        provider = self.find_api_provider(resources)
-        provider.extract_resources(resources, collector, cwd=self.cwd)
+        provider = self.find_api_provider(self.stacks)
+        provider.extract_resources(self.stacks, collector, cwd=self.cwd)
         return collector.get_api()
 
     @staticmethod
-    def find_api_provider(resources):
+    def find_api_provider(stacks: List[Stack]):
```

Comment:
The logic here seems off. Different API resources my have different providers. Should this return a resource-provider map?

Reply:
I did not change much for this method. What seems off here?

Reply:
I understand, I'm saying the whole logic here seems confusing. It seems like we don't really care which provider. Anyway, this is out of scope of this PR

Reply:
yeah, agree, it is confusing. 

The behavior is:
1. go through all resources until find an API.
2. use the API type to determine the whole start-api behavior
    - if the first found api is  SAM api, we are gonna ignore all CFN APIs
    - vice versa